app/models/user.rb

```diff
@@ -1,4 +1,6 @@
 class User < ApplicationRecord
+  after_create :add_global_viewer_group, if: :is_global_viewer?
```

Comment:
please create a model spec that make sure global viewer group is added if global viewer configuration is active

Reply:
done